### PR TITLE
test: make sure tests fail properly instead of timing out

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Makes the tests fail properly instead of timing out:
- either wrap `expect()` in `try` / `catch`
- or re-implement test with `async` / `await`

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes
Notes: no-notes

### CODE REVIEW

File: spec-main/chromium-spec.ts [hunk 1]
```diff
@@ -561,12 +561,11 @@ describe('chromium features', () => {
...
-        w.webContents.once('new-window', (e, url, frameName, disposition, options) => {
-          expect(options.show).to.equal(w.isVisible());
-          w.close();
-          done();
-        });
+        const newWindow = emittedOnce(w.webContents, 'new-window');
         w.loadFile(path.join(fixturesPath, 'pages', 'window-open.html'));
+        const [,,,, options] = await newWindow;
+        expect(options.show).to.equal(w.isVisible());
+        w.close();
```
Comment (alexeykuzmin) [inline spec-main/chromium-spec.ts:568]: Won't be closed in the expectation fails.

File: spec-main/node-spec.ts [hunk 2]
```diff
@@ -12,13 +12,12 @@ describe('node feature', () => {
...
+      it('Works in browser process', async () => {
         const child = childProcess.fork(path.join(fixtures, 'module', 'ping.js'));
-        child.on('message', (msg) => {
-          expect(msg).to.equal('message');
-          done();
-        });
+        const messageEvent = emittedOnce(child, 'message');
         child.send('message');
+        const [msg] = await messageEvent;
+        expect(msg).to.equal('message');
```
Comment (alexeykuzmin) [inline spec-main/node-spec.ts:20]: `msg` -> `message`

File: spec-main/chromium-spec.ts (hunk 1 above)
Comment (nornagon) [reply spec-main/chromium-spec.ts:568]: https://github.com/electron/electron/pull/24019 fyi

File: spec-main/node-spec.ts (hunk 2 above)
Comment (miniak) [reply spec-main/node-spec.ts:20]: why?

File: spec/webview-spec.js [hunk 3]
```diff
@@ -1045,11 +1058,15 @@ describe('<webview> tag', function () {
     it('does not emit when src is not changed', (done) => {
       loadWebView(webview);
       setTimeout(() => {
```
Comment (alexeykuzmin) [inline spec/webview-spec.js:1060]: `await delay()`?

File: spec/webview-spec.js [hunk 4]
```diff
@@ -803,8 +812,12 @@ describe('<webview> tag', function () {
       server.listen(0, '127.0.0.1', () => {
         const port = server.address().port;
         webview.addEventListener('ipc-message', (e) => {
```
Comment (alexeykuzmin) [inline spec/webview-spec.js:814]: `waitForEvent()`?

File: spec/chromium-spec.js [hunk 5]
```diff
@@ -385,9 +450,14 @@ describe('chromium feature', () => {
           // wherein calling `getItem` immediately after `setItem` would appear to work
           // but then later (e.g. next tick) it would not.
           setTimeout(() => {
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js:452]: `await delay(1)`?

File: spec/chromium-spec.js [hunk 6]
```diff
@@ -331,9 +387,14 @@ describe('chromium feature', () => {
           console.log(e);
         });
         webview.addEventListener('ipc-message', (e) => {
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js:389]: `waitForEvent()`?

File: spec/chromium-spec.js [hunk 7]
```diff
@@ -311,17 +359,25 @@ describe('chromium feature', () => {
...
+          } catch (e) {
+            done(e);
+          }
         };
         worker.port.postMessage(message);
       });
 
       it('has no node integration by default', (done) => {
         const worker = new SharedWorker('../fixtures/workers/shared_worker_node.js');
         worker.port.onmessage = (event) => {
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js:374]:
```js
const event = await new Promise((resolve) => worker.port.onmessage = resolve);
```

File: spec/chromium-spec.js [hunk 8]
```diff
@@ -277,28 +310,43 @@ describe('chromium feature', () => {
...
+          done(e);
+        } finally {
+          worker.terminate();
+        }
       };
     });
 
     it('Worker has node integration with nodeIntegrationInWorker', (done) => {
       const webview = new WebView();
       webview.addEventListener('ipc-message', (e) => {
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js:341]: `waitForEvent()`?

File: spec/chromium-spec.js [hunk 9]
```diff
@@ -277,28 +310,43 @@ describe('chromium feature', () => {
...
     });
 
     it('Worker has no node integration by default', (done) => {
       const worker = new Worker('../fixtures/workers/worker_node.js');
       worker.onmessage = (event) => {
-        expect(event.data).to.equal('undefined undefined undefined undefined');
-        worker.terminate();
-        done();
+        try {
+          expect(event.data).to.equal('undefined undefined undefined undefined');
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js:329]:
```js
const event = await new Promise((resolve) => worker.onmessage = resolve);
worker.terminate;
expect(...);
```

File: spec/webview-spec.js (hunk 4 above)
Comment (miniak) [reply spec/webview-spec.js:814]: it does not help much here

**Pull request merged**